dxr/static/js/dxr.js

```diff
@@ -423,19 +408,32 @@ $(function() {
         hideBubble();
         nextRequestNumber += 1;
         oneMoreRequest();
-        $.getJSON(buildAjaxURL(query, caseSensitiveBox.prop('checked'), limit, 0), function(data) {
-            data.query = query;
-            // New results, overwrite
-            if (myRequestNumber > displayedRequestNumber) {
-                displayedRequestNumber = myRequestNumber;
-                populateResults(data, false);
-                historyWaiter = setTimeout(pushHistoryState, timeouts.history, data);
-            }
+        $.ajax({
+            dataType: "json",
+            url: queryString,
+            // We need to disable caching of this result because otherwise we break the undo close
+            // tab feature on search pages (Chrome and Firefox).
+            cache: false,
+            success: function (data) {
+                // Check whether to redirect to a direct hit.
+                if (data.redirect) {
+                    window.location.href = data.redirect;
+                    return;
+                }
+                data.query = query;
+                // New results, overwrite
+                if (myRequestNumber > displayedRequestNumber) {
+                    displayedRequestNumber = myRequestNumber;
+                    populateResults(data, false);
+                    historyWaiter = setTimeout(history.pushState.bind(history, {}, '', queryString),
```

Comment:
If the user hit Return (i.e. redirect is true), let's update the URL immediately instead of waiting.